setup.py

```diff
@@ -1,109 +1,192 @@
 #!/usr/bin/python
-from setuptools import setup, Extension
 from Cython.Build import cythonize
+from setuptools import setup, Extension, Distribution
+from subprocess import call, PIPE, Popen
 import os
-import sys
-import subprocess
 import re
+import shlex
+import subprocess
+import sys
 
-hidapi_topdir = os.path.join("hidapi")
-hidapi_include = os.path.join(hidapi_topdir, "hidapi")
-system_hidapi = 0
-libs = []
-src = ["hid.pyx"]
+min_required_hidapi_version = '0.14'
 
+libusb_pkgconfig = 'libusb-1.0 >= 1.0.9'
+hidapi_libusb_pkgconfig = 'hidapi-libusb >= ' + min_required_hidapi_version
+hidapi_hidraw_pkgconfig = 'hidapi-hidraw >= ' + min_required_hidapi_version
+hidapi_pkgconfig = 'hidapi >= ' + min_required_hidapi_version
 
-def hidapi_src(platform):
-    return os.path.join(hidapi_topdir, platform, "hid.c")
+tld = os.path.abspath(os.path.dirname(__file__))
+embedded_hidapi_topdir = os.path.join(tld, "hidapi")
+embedded_hidapi_include = os.path.join(embedded_hidapi_topdir, "hidapi")
 
 
-if "--with-system-hidapi" in sys.argv:
-    sys.argv.remove("--with-system-hidapi")
-    system_hidapi = 1
-    hidapi_include = "/usr/include/hidapi"
+def get_extension_compiler_type():
+    """Returns a compiler to be used by setuptools to build Extensions
+        Taken from https://github.com/pypa/setuptools/issues/2806#issuecomment-961805789
+    """
+    d = Distribution()
+    build_ext = Distribution().get_command_obj("build_ext")
+    build_ext.finalize_options()
+    # register an extension to ensure a compiler is created
+    build_ext.extensions = [Extension("ignored", ["ignored.c"])]
+    # disable building fake extensions
+    build_ext.build_extensions = lambda: None
+    # run to populate self.compiler
+    build_ext.run()
+    return build_ext.compiler.compiler_type
+
+
+# this could have been just pkgconfig.configure_extension(), but: https://github.com/matze/pkgconfig/issues/65
+# additionally contains a few small improvements
+def pkgconfig_configure_extension(ext, package):
+    pkg_config_exe = os.environ.get('PKG_CONFIG', None) or 'pkg-config'
+
+    def exists(package):
+        cmd = "{0} --exists '{1}'".format(pkg_config_exe, package)
+        return call(shlex.split(cmd)) == 0
+
+    if not exists(package):
+        raise Exception(f"pkg-config package '{package}' not found")
+
+    def query_pkg_config(package, *options):
+        cmd = "{0} {1} '{2}'".format(pkg_config_exe, ' '.join(options), package)
```

Comment:
Let's use f-strings:


```suggestion
        cmd = "{pkg_config_exe} {' '.join(options)} '{package}'"
```